File: src/main/java/com/optica/services/CurriculumServiceImpl.java
package com.optica.services;



import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.optica.domain.Curriculum;
import com.optica.repository.CurriculumDao;

@Service
public class CurriculumServiceImpl implements CurriculumService {

	@Autowired
	private CurriculumDao curriculumDao;

	@Override
	public Curriculum findById(Long id) {
		return curriculumDao.findById(id).orElseThrow(() -> new IllegalArgumentException("Curriculum not found"));
	}

	@Override
	public void save(Curriculum curriculum) {
		curriculumDao.save(curriculum);

	}

	@Override
	public void delete(Long id) {
		curriculumDao.deleteById(id);

	}

}

File: src/main/kotlin/com/optica/services/CurriculumServiceImpl.kt
package com.optica.services


import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

import com.optica.domain.Curriculum
import com.optica.repository.CurriculumDao

@Service
class CurriculumServiceImpl : CurriculumService {

    @Autowired
    private val curriculumDao: CurriculumDao? = null

    override fun findById(id: Long?): Curriculum {
        return curriculumDao!!.findById(id!!).orElse(Curriculum())
    }

    override fun save(curriculum: Curriculum) {
        curriculumDao!!.save(curriculum)

    }

    override fun delete(id: Long?) {
        curriculumDao!!.deleteById(id!!)

    }

}
